### ISSUE TITLE
Fixture breaks when defined in class method

### ISSUE THREAD

**Issue by mcstarioni** [smarie/python-pytest-cases#182]
Using pytest_cases's fixture in a test which is a class method leads to the following error: `fixture 'self' not found`,. While using pytest.fixture doesn't result in error. 
Minimal code to reproduce:
```
import pytest
from pytest_cases import fixture


class TestMethod:
    @pytest.fixture
    def pytest_fxt(self):
        yield "Hello"

    def test_with_pytest(self, pytest_fxt):
        # succeeds
        assert pytest_fxt == "Hello"

    @fixture
    def cases_fxt(self):
        return "Hello"

    def test_with_cases(self, cases_fxt):
        # raises an error with regards to 'self'
        assert cases_fxt == "Hello"

```

Versions: pytest: '6.2.1', pytest_cases: '3.2.0'

**Comment by smarie** [comment]
Thanks @mcstarioni for reporting this issue, sorry for not having seen the notification :(
I'll have a look

**Comment by smarie** [comment]
Indeed. The reason seems to be that I was adding a `request` dependency as first argument as it should be last

```python
TestMethod.pytest_fxt
Out[2]: <function pytest_cases.tests.pytest_extension.issues.test_issue_182.TestMethod.pytest_fxt(self)>
TestMethod.cases_fxt
Out[3]: <function pytest_cases.tests.pytest_extension.issues.test_issue_182.TestMethod.cases_fxt(request, self)>
```

This additional argument is used when this fixture is used in a union: pytest will always try to call it even if it is not used in this branch of the union, so I create a wrapper inspection the `request` to detect if it should *really* be called or not.

**Comment by smarie** [comment]
Fixed in version 3.2.1 that should be available in a few minutes when travis has completed. Let me know !
